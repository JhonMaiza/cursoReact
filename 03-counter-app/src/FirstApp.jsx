import PropTypes from 'prop-types';

// si no hay dependencia de los componente se debe sacar del funcional component
const funcion = (a, b) => {
  console.log('hola');
  return a + b;
}

const newMessage = {
  message: 'hola',
  titulo: 'Jhon'
}

export const FirstApp = ( { title, subtitle, nombre} ) => {
  
  return (
    <>
      {/* <code>{ JSON.stringify(newMessage)</code>  } */}
      <h1 data-testid="test-title">{ title }</h1>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p data-testid="test-subtitle">{subtitle}</p>
      <p>{nombre}</p>
      {/* {funcion(2, 3)} */}
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number.isRequired
}

// defaul van antes de los propTypes
// FirstApp.defaultProps = {
//   title: 'No hay t√≠tulo',
//   subtitle: 'No hay subtitulo',
//   nombre: 'Pedro'
// }
