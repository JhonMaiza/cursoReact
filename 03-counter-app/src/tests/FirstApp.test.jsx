import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp";

describe('Pruebas en < FirstApp />', () => {
  const title = 'Hola soy Goku';
  const subtitle = 123;

  test('debe de hacer match con el snapshot', () => {
    const { container } = render( <FirstApp title={title} subtitle={ subtitle }/>);
    expect( container ).toMatchSnapshot();
  })

  test('debe mostar el mensaje "Hola, Soy Goku', () => {
    render(<FirstApp title={title} subtitle={ subtitle }/>);
    expect( screen.getByText( title )).toBeTruthy();
  });

  test('debe de mostrar el titulo en un h1', () => {
    render(<FirstApp title={ title } subtitle={ subtitle }/>);
    expect( screen.getByRole( 'heading', { level:1 } ).innerHTML).toContain(title);
  });
  
  test('debe de mostrar el subtitulo enviado por props', () => {
    render(
      <FirstApp 
        title={ title } 
        subtitle={ subtitle }
      />
    );
    expect( screen.getAllByText(subtitle).length).toBe(2);
  });
  
  
  
})
