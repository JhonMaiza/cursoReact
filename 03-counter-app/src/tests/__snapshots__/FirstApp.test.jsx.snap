// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Goku
  </h1>
  <p
    data-testid="test-subtitle"
  >
    123
  </p>
  <p
    data-testid="test-subtitle"
  >
    123
  </p>
  <p />
</div>
`;
